python/nano/src/bigdl/nano/pytorch/strategies/ddp_spawn.py

```diff
@@ -221,6 +227,68 @@ def setup(self, trainer: "pl.Trainer") -> None:
             # so we need to set it to `False` manuallay
             self.model.eval()
 
+        if trainer.training and self.auto_lr:
+
+            def _unpack_lightning_optimizer(opt):
+                return opt._optimizer if isinstance(opt, LightningOptimizer) else opt
+
+            optimizers = self.optimizers
+            optimizers = [_unpack_lightning_optimizer(opt) for opt in optimizers]
+
+            for optimizer in optimizers:
+                for param_group in optimizer.param_groups:
+                    param_group["lr"] *= self.world_size
+
+            lr_scheduler_configs = self.lr_scheduler_configs
+            for config in lr_scheduler_configs:
+                scheduler = config.scheduler
+                scheduler.base_lrs = [lr * self.world_size for lr in scheduler.base_lrs]
+
+            if self.lr_scheduler_configs:
+                warnings.warn(f"Nano warmup currently only support none optimizers, "
+                              f"but got {len(self.lr_scheduler_configs)}. Skip warmup")
+            else:
+                lr_schedulers = []
+                for opt_idx, opt in enumerate(self.optimizers):
+                    if type(self.auto_lr) is bool:
+                        self.auto_lr = trainer.max_epochs // 5 + 1
```

Comment:
Is there any reference or reasoning for setting this strategy?

Reply:
Actually, I didn't find a specific suggestion on how to set warm_epochs. The warmup_epochs is 4000 and max_epochs is 40000 in some transformer implemetation. In facebook's paper, warmup_epoch is set to 5 and max_epochs is 80.